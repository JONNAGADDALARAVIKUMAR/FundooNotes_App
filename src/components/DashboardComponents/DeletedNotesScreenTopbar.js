import React, { Component } from 'react';
import {View, Text} from 'react-native';
import {strings} from '../../Languages/strings';
import {Appbar} from 'react-native-paper';
import DeletedNotesScreenStyles from '../../styles/DeletedNotesScreenStyles';

export default class DeletedNotesScreenTopbar extends Component {
    constructor(props) {
        super(props);
    }

    openDrawer = () => {
        const {onPress} = this.props
        this.props.navigation.openDrawer()
        onPress();
    }

    render() {
        return(
            <View>
                <Appbar style = {DeletedNotesScreenStyles.App_Bar_Style}>
                    <Appbar.Action
                        icon="menu"
                        onPress = {this.openDrawer}
                    />
                    <Text style = {DeletedNotesScreenStyles.Deleted_Text_Style}>{strings.Deleted}</Text>
                    <Appbar.Content/>
                    <Appbar.Action 
                        icon = "dots-vertical"
                    />
                </Appbar>
            </View>
        )
    }
}