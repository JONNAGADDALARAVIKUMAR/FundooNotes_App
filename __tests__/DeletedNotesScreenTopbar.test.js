import React from 'react';
import {shallow} from 'enzyme';
import DeletedNotesScreenTopbar from '../src/components/DashboardComponents/DeletedNotesScreenTopbar';

describe('test DeletedNotesScreenTopbar Class', () => {
    it('test when render should match to snapshot', async () => {
        const component = shallow(<DeletedNotesScreenTopbar/>)
        expect(component).toMatchSnapshot();
    })

    it('test onPress event of DeletedNotesScreen it should navigate to AddNewNotes Screen', async() => {
        const navigation = { openDrawer : jest.fn() }
        const onPressEvent = jest.fn();
        const component = shallow(<DeletedNotesScreenTopbar onPress = {onPressEvent} navigation = {navigation} />)
        const instance = component.instance();
        await instance.openDrawer();
        expect(onPressEvent).toHaveBeenCalled();
        expect(navigation.openDrawer).toBeCalled()
    })
})