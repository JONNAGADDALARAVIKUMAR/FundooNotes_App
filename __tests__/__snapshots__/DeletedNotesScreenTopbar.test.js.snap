// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test DeletedNotesScreenTopbar Class test when render should match to snapshot 1`] = `ShallowWrapper {}`;
